### PR TITLE
move quadlet packages into pkg/systemd

### PR DESCRIPTION
**Pull request by vrothberg**
Reduce the number of top-level packages in ./pkg by moving quadlet packages under ./pkg/systemd.

[NO NEW TESTS NEEDED] - no functional change.

Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```
@alexlarsson @rhatdan PTAL

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: cmd/quadlet/main.go [hunk 1]
```diff
@@ -9,8 +9,8 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/containers/podman/v4/pkg/quadlet"
-	"github.com/containers/podman/v4/pkg/systemdparser"
+	systemdparser "github.com/containers/podman/v4/pkg/systemd/parser"
```
Comment (Luap99) [inline cmd/quadlet/main.go:12]: Can we not rename this? I do not really like import renames unless required to name conflict.

Comment (Luap99) [reply cmd/quadlet/main.go:12]: `"github.com/containers/podman/v4/pkg/systemd/parser"` is fine for me, just do not rename the package here.

**Review by Luap99 — COMMENTED** [review]
LGTM

**Pull request merged**